jenkins/jenkinsfiles/Jenkinsfile-nightly.groovy

```diff
@@ -0,0 +1,116 @@
+		booleanParam name: shouldPublishFailJobStatusName, description: 'true to publish job status if failed', defaultValue: true
+		booleanParam name: 'WAIT_FOR_BUILDS', description: 'true to wait for trigger build to complete', defaultValue: true
+	}
+
+	agent {
+		label 'ubuntu-agent'
+	}
+
+	triggers {
+		cron('@midnight')
+	}
+
+	options {
+		ansiColor('css')
+		timestamps()
+	}
+
+	stages {
+		stage('display environment') {
+			steps {
+				sh 'printenv'
+			}
+		}
+		stage('trigger build jobs') {
+			steps {
+				script {
+					triggerAllJobs(
+						env.JOB_BRANCH_NAME ?: defaultBranch,
+						env.WAIT_FOR_BUILDS ? env.WAIT_FOR_BUILDS.toBoolean() : true,
+						shouldPublishFailJobStatusName,
+						env.SHOULD_PUBLISH_FAIL_JOB_STATUS ? env.SHOULD_PUBLISH_FAIL_JOB_STATUS.toBoolean() : true
+					)
+				}
+			}
+		}
+	}
+	post {
+		unsuccessful {
+			script {
+				if (null != env.SHOULD_PUBLISH_FAIL_JOB_STATUS && env.SHOULD_PUBLISH_FAIL_JOB_STATUS.toBoolean()) {
+					helper.sendDiscordNotification(
+						"Jenkins Job Failed for ${currentBuild.fullDisplayName}",
+						"At least one job failed for Build#${env.BUILD_NUMBER} which has a result of ${currentBuild.currentResult}.",
+						env.BUILD_URL,
+						currentBuild.currentResult
+					)
+				}
+			}
+		}
+	}
+}
+
+void triggerAllJobs(
+		String branchName,
+		boolean waitForDownStream,
+		String shouldPublishFailJobStatusName,
+		boolean shouldPublishFailJobStatusValue) {
+	Map<String, String> siblingNameMap = siblingJobNames()
+	Map<String, Closure> buildJobs = [:]
+	final String platformName = 'PLATFORM'
+
+	siblingNameMap.each { siblingName ->
+		String displayName = siblingName.value
+		buildJobs["${displayName}"] = {
+			stage("${displayName}") {
+				String fullJobName = siblingName.key + '/' + branchName
+
+				// Platform parameter can vary per project, get the last value
+				echo "job name - ${fullJobName}"
+				String platform = lastJobParameterValue(fullJobName, platformName)
+
+				build job: "${fullJobName}", parameters: [
+					gitParameter(name: 'MANUAL_GIT_BRANCH', value: branchName),
+					string(name: platformName, value: platform ?: 'ubuntu'),
+					string(name: 'BUILD_CONFIGURATION', value: 'release-private'),
+					string(name: 'TEST_MODE', value: 'code-coverage'),
+					booleanParam(name: 'SHOULD_PUBLISH_IMAGE', value: false),
+					booleanParam(name: shouldPublishFailJobStatusName, value: shouldPublishFailJobStatusValue)],
+					wait: waitForDownStream
+			}
+		}
+	}
+
+	parallel buildJobs
+}
+
+Map<String, String> siblingJobNames() {
+	Item project = Jenkins.get().getItemByFullName(currentBuild.fullProjectName)
+	List<Item> siblingItems = project.parent.items
+
+	Map<String, String> targets = [:]
+	for (Item item in siblingItems) {
+		// groovylint-disable-next-line Instanceof
+		if (!(item instanceof hudson.model.AbstractModelObject) || item.fullName == project.fullName) {
```

Comment:
can you check if it has property fullName instead of instanceOf check ?

Reply:
Yea, let me take a look.  might be able to just remote this check also.